src/bind.c

```diff
@@ -68,6 +76,23 @@ static SEXP vec_rbind(SEXP xs, SEXP ptype, enum name_repair_arg name_repair) {
   SEXP idx = PROTECT(compact_seq(0, 0));
   int* idx_ptr = INTEGER(idx);
 
+  SEXP names_to_col = R_NilValue;
+  SEXP* names_to_p = NULL;
+  const SEXP* xs_names_p = NULL;
+
+  int nprot = 0;
+  if (names_to != R_NilValue) {
+    SEXP xs_names = PROTECT_N(r_names(xs), &nprot);
+
+    if (!r_is_names(xs_names)) {
+      Rf_errorcall(R_NilValue, "Inputs must have names when `.names_to` is supplied");
```

Comment:
In other places we've relaxed this condition to use the index when there are no names.

Reply:
hmm, then does it matter that `names_to` refers to names? `index_to` doesn't seem right either.

Reply:
I think it's ok.